services/api.service.ts

```diff
@@ -151,15 +151,31 @@ export default class ApiService extends moleculer.Service {
 
     let user: User;
     if (authUser.type === AuthUserRole.USER) {
-      user = await ctx.call('users.findOne', {
-        query: { authUser: authUser.id },
-      });
+        user = await ctx.call('users.findOne', {
+            query: {
+                authUser: authUser.id
+            },
+        });
+        const profile = req.headers['x-profile'] as any
+        if (!!profile && !isNaN(profile)) {
```

Comment:
default `isNaN` has same issues - https://www.geeksforgeeks.org/lodash-_-isnan-method/. Maybe it would be better to not check here (`isNaN`) and leave it like this. Just because if you pass header `X-Profile: 'hello'` it will not check autorization, right?